### PR TITLE
ZK-4032: zul.sel.ItemIter will return non-item object if all items ar…

### PR DESCRIPTION
**Pull request by klyvechen**
…e visible(false) during init

### CODE REVIEW

File: zul/src/archive/web/js/zul/sel/Listbox.js [hunk 1]
```diff
@@ -506,7 +506,7 @@ zul.sel.ItemIter = zk.$extends(zk.Object, {
 			this._isInit = true;
 			var p = this.box.firstItem;
 			if (this.opts && this.opts.skipHidden)
-				for (; p && !p.isVisible(); p = p.nextSibling) {}
+				for (; p && !p.isVisible(); p = p.parent.nextItem(p)) {}
```
Comment (scribetw) [inline zul/src/archive/web/js/zul/sel/Listbox.js]: `p = this.box.nextItem(p)`

File: zkdoc/release-note [hunk 2]
```diff
@@ -55,6 +55,7 @@ Aug 28, 2018
   ZK-4021: TypeError: ref is undefined after ZK-3943
   ZK-3987: a label in a cell has a different default color
   ZK-4004: Invisible columns affect mesh column span width count with flex
+  ZK-4032: zul.sel.ItemIter will return non-item object if all items are visible(false) during init
```
Comment (wenninghsu) [inline zkdoc/release-note]: 8.6.0

File: zktest/src/archive/test2/B86-ZK-4032.zul [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+<zk>
+	<zscript><![CDATA[
+   		List model = new ArrayList();
+   		model.add("a");
+   		model.add("b");
+   		model.add("c");
+   		model.add("d");
+	]]></zscript>
+	<label multiline="true">
+		1. Connect to the page,
```
Comment (wenninghsu) [inline zktest/src/archive/test2/B86-ZK-4032.zul]: already here

Comment (klyvechen) [reply zktest/src/archive/test2/B86-ZK-4032.zul]: ok

File: zkdoc/release-note (hunk 2 above)
Comment (klyvechen) [reply zkdoc/release-note]: ok

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**